src/main/java/seedu/address/logic/commands/exceptions/CommandException.java

```diff
@@ -23,21 +23,58 @@ public CommandException(String message, Throwable cause) {
-         * This is to be used for user interaction.
-         *
-         * @param company
+         * This is to be used for user interaction for the EDIT command.
          */
-        public DuplicateCompanyException(Company company) {
+        public DuplicateCompanyException(Company company, int index) {
             super("Duplicate Entry Detected"
                     + "\n" + "You already have another entry "
                     + "for the company " + company.getName() + " with the role "
-                    + company.getRole() + "."
-                    + "\n" + "Here are the details of the existing entry: "
-                    + "\n" + company.duplicatedCompanyFlagOutput()
+                    + company.getRole() + " at index " + (index + 1) + "."
             );
+            expectedMessage = "Duplicate Entry Detected"
+                    + "\n" + "You already have another entry "
+                    + "for the company " + company.getName() + " with the role "
+                    + company.getRole() + " at index " + (index + 1) + ".";
+        }
+
+        /**
+         * Constructs a new {@code DuplicateCompanyException} with the specified detail {@code message}.
+         * This is to be used for user interaction for the ADD Command.
+         */
+        public DuplicateCompanyException(Company company, int index, String allChangedFields) {
+            super(allChangedFields.isEmpty()
+                    ? ("Duplicate Entry Detected"
+                    + "\n" + "You already have another entry with the exact same details "
+                    + "for the company " + company.getName() + " with the role "
+                    + company.getRole() + " at index " + (index + 1) + ".") : ("Duplicate Entry Detected"
+                    + "\n" + "You already have another entry "
+                    + "for the company " + company.getName() + " with the role "
+                    + company.getRole() + " at index " + (index + 1) + "."
+                    + "\n" + "Perhaps you meant to use the edit command instead?"
+                    + " Type: " + "edit " + (index + 1) + " " + allChangedFields)
+            );
+            expectedMessage = allChangedFields.isEmpty()
+                    ? ("Duplicate Entry Detected"
+                            + "\n" + "You already have another entry with the exact same details "
+                            + "for the company " + company.getName() + " with the role "
+                            + company.getRole() + " at index " + (index + 1) + ".") : ("Duplicate Entry Detected"
+                    + "\n" + "You already have another entry "
+                    + "for the company " + company.getName() + " with the role "
+                    + company.getRole() + " at index " + (index + 1) + "."
+                    + "\n" + "Perhaps you meant to use the edit command instead?"
+                    + " Type: " + "edit " + (index + 1) + " " + allChangedFields);
+        }
```

Comment:
is there a reason for the duplicated message here? 

could we put this into Messages, and call it instead?

Reply:
Yes, not possible to call intermediate because of `super`. I can move this message to Messages but still, will does not solve the duplicated code problem.